internal/util/update/update_test.go

```diff
@@ -703,6 +704,79 @@ func TestCommand_Run_failInvalidPath(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_badUpstreamLock(t *testing.T) {
+	testCases := map[string]struct {
+		dir              string
+		commit           string
+		repo             string
+		expectedErrorMsg string
+	}{
+		"bad commit": {
+			dir:              "/",
+			commit:           "fake",
+			repo:             testutil.Upstream,
+			expectedErrorMsg: "unknown revision or path"},
+		"bad dir": {
+			dir:              "/fake",
+			repo:             testutil.Upstream,
+			expectedErrorMsg: "no such file or directory"},
+		"bad repo": {
+			dir:              "/",
+			repo:             "fake",
+			expectedErrorMsg: "does not appear to be a git repository"},
+	}
+	for tn, tc := range testCases {
+		t.Run(tn, func(t *testing.T) {
+			upstreamContent := testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithResource(pkgbuilder.DeploymentResource),
+				Branch: masterBranch,
+			}
+
+			g := &testutil.TestSetupManager{
+				T: t,
+				ReposChanges: map[string][]testutil.Content{
+					testutil.Upstream: {
+						upstreamContent,
+					},
+				},
+			}
+
+			defer g.Clean()
+
+			if !g.Init() {
+				return
+			}
+
+			// If tc.repo exists in the testbed, use its repo directory for the UpstreamLock.
+			repo := g.Repos[tc.repo]
+			repoDir := tc.repo
+			if repo != nil {
+				repoDir = repo.RepoDirectory
+			}
+			localChanges := []testutil.Content{{
+				Pkg: pkgbuilder.NewRootPkg().WithKptfile(pkgbuilder.NewKptfile().
+					WithUpstreamRef(testutil.Upstream, tc.dir, masterBranch, "resource-merge").
```

Comment:
We should probably avoid using the `tc.dir` value here so we know that the `Upstream` section is correct.

Reply:
Done, added a check in `update.go` to throw an error on faulty `UpstreamLock` paths.